File: phoxi/src/grab_frame.cpp
#include "phoxi_sensor.h"

int main()
{
    PhoxiSensor sensor("phoxi", "2018-02-020-LC3", "extra-large");
    sensor.start();
    sensor.frames();

    std::vector<std::vector<float>> depth_map = sensor.get_depth_map(); 

    sensor.stop();

    return 0;
}

File: phoxi/examples/grab_frame.py
import numpy as np
import matplotlib.pyplot as plt

import sys
sys.path.append('/home/andreas/Software/perception/phoxi/build/')

from phoxi import PhoxiSensor

# import ipdb; ipdb.set_trace()

sensor = PhoxiSensor('', '', '')
sensor.start()
sensor.frames()
depth_map = sensor.get_depth_map()
depth_map = np.array(depth_map)

plt.imshow(depth_map)
plt.show()
